lib/nimble_template/helpers/credo.ex

```diff
@@ -1,10 +1,75 @@
 defmodule NimbleTemplate.CredoHelper do
   alias NimbleTemplate.Generator
+  alias NimbleTemplate.Projects.Project
+
+  @do_single_expression_rule_name "CompassCredoPlugin.Check.DoSingleExpression"
+  @single_module_file_rule_name "CompassCredoPlugin.Check.SingleModuleFile"
 
   @spec disable_rule(String.t(), String.t()) :: :ok | {:error, :failed_to_read_file}
   def disable_rule(file_path, rule) do
-    Generator.prepend_content(file_path, """
-    # credo:disable-for-this-file #{rule}
-    """)
+    if File.exists?(file_path) do
+      Generator.prepend_content(file_path, """
+      # credo:disable-for-this-file #{rule}
+      """)
+    end
+  end
+
+  @spec disable_do_single_expression_rule(list()) :: :ok | {:error, :failed_to_read_file}
+  def disable_do_single_expression_rule(file_paths) do
+    Enum.each(file_paths, fn file_path ->
+      disable_rule(file_path, @do_single_expression_rule_name)
+    end)
+  end
+
+  @spec disable_single_module_file_rule(list()) :: :ok | {:error, :failed_to_read_file}
+  def disable_single_module_file_rule(file_paths) do
+    Enum.each(file_paths, fn file_path ->
+      disable_rule(file_path, @single_module_file_rule_name)
+    end)
+  end
```

Comment:
As `disable_rule`, `disable_do_single_expression_rule` and `disable_single_module_file_rule` functions are not called from outside of this module and only from functions below shall we move it to towards the end and make them private? Or do you think there's plans for it to be called from outside the module in the future?

I know this existed on disable_rule already but do we need to add `@spec` to these? I think we don't use this consistently throughout the template project or is there only specific cases where we use it?

Reply:
Good suggestion, refactored in https://github.com/nimblehq/elixir-templates/pull/308/commits/ce27c66a70f9c83889b6fe91e171cd5a03d244e4